Nemo11235/CS160-Final-Project
# Generate room

This PR does the following: 
- Added the user id and room id rules on multiplayer page;
- Added a function that validates room id;
- Room id can now be auto-generated;
- One-click copy button to copy the generated room id;
- More detailed feedback on invalid input of user id and room id;


QA Steps:
1. npm start on the server;
2. Go to the multiplayer page;
3. The Rules should show on top:
![image](https://user-images.githubusercontent.com/26106407/166124773-c4a7e9ce-7c08-4f3a-b04f-5e71a53f45cd.png)
4. Click on the `Generate Room ID` button, a 6 letter/number long room id should be generated;
![image](https://user-images.githubusercontent.com/26106407/166124796-656d124f-4340-4aaf-a164-56f101634fcb.png)
5. Click on the copy icon on the right, the id should be copied to the clipboard red text `copied!` should show at the bottom. 
![image](https://user-images.githubusercontent.com/26106407/166124965-d71554a6-7e47-40b0-bc49-451c6f41c0e1.png)
6. Click on the input box of Room id, click `ctrl` + `v` to paste the room id;
7. Enter a valid User ID;
8. Click on the `Join` button, user should be able to join the game;
Now we test on invalid Room ID. Enter a valid User ID, then:
1. For Room ID, enter "???" or any special character and click Join, alert should show:
![image](https://user-images.githubusercontent.com/26106407/166124924-9ff85d9e-ba30-4e1a-9b07-46927f51e5c4.png)
2. For Room ID, enter "123456789" or any string longer than 6, and click Join, alert should show:
![image](https://user-images.githubusercontent.com/26106407/166124947-e96d673f-f865-4b6e-a109-3822415eab9a.png)
